test/support/factories.ex

```diff
@@ -212,11 +247,63 @@ defmodule Lightning.Factories do
       )
 
     build(:workflow, attrs)
-    |> with_job(job)
     |> with_trigger(trigger)
     |> with_edge({trigger, job})
   end
 
+  def complex_workflow_factory(attrs) do
+    #   
+    #          +---+
+    #          | T |    
+    #          +---+    
+    #            |      
+    #            |      
+    #          +---+
+    #      +---- 0 ----+
+    #      |   +---+   |
+    #      |           |
+    #      |           |
+    #      |           |
+    #    +-|-+       +-|-+
+    #    | 1 |       | 4 |
+    #    +-|-+       +-|-+
+    #      |           |
+    #      |           |
+    #    +-+-+       +-+-+
+    #    | 2 |       | 5 |
+    #    +-|-+       +-|-+
+    #      |           |
+    #      |           |
+    #    +-+-+       +-+-+
+    #    | 3 |       | 6 |
+    #    +---+       +---+
+    #   
+
+    trigger = build(:trigger, type: :webhook)
+
+    jobs =
+      build_list(7, :job,
+        name: fn -> sequence(:name, &"Job-#{&1}") end,
+        body: ~s[fn(state => { return {...state, extra: "data"} })],
+        workflow: nil
+      )
+
+    build(:workflow, attrs)
+    |> with_trigger(trigger)
+    |> then(fn workflow ->
+      Enum.reduce(jobs, workflow, fn job, workflow ->
+        workflow |> with_job(job)
+      end)
+    end)
+    |> with_edge({trigger, jobs |> Enum.at(0)})
+    |> with_edge({jobs |> Enum.at(0), jobs |> Enum.at(1)})
+    |> with_edge({jobs |> Enum.at(1), jobs |> Enum.at(2)})
+    |> with_edge({jobs |> Enum.at(2), jobs |> Enum.at(3)})
+    |> with_edge({jobs |> Enum.at(0), jobs |> Enum.at(4)})
+    |> with_edge({jobs |> Enum.at(4), jobs |> Enum.at(5)})
+    |> with_edge({jobs |> Enum.at(5), jobs |> Enum.at(6)})
+  end
```

Comment:
Here is the replacement for `JobFixtures.workflow_scenario/1`, it's return value is a little different - in that it's the Workflow (and not a map), you're expected to destructure/match jobs/triggers etc from it.

`workflow_scenario/1` has been marked as deprecated, there will be warnings from the compiler. We need to replace all calls to workflow_scenario with the new factory.